api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3939,6 +3940,129 @@ public void setBucketPolicy(String bucketName, String policy)
+   *                                     InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException         upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidArgumentException    upon invalid value is passed to a method.
+   */
+  public void setBucketLifeCycle(String bucketName, String lifeCycle)
+          throws InvalidBucketNameException, NoSuchAlgorithmException,
+          InsufficientDataException, IOException, InvalidKeyException, NoResponseException,
+          XmlPullParserException, ErrorResponseException, InternalException,InvalidArgumentException {
+    if ((lifeCycle == null) || "".equals(lifeCycle)) {
+      throw new InvalidArgumentException("life cycle cannot be empty");
+    }
+    Map<String, String> headerMap = new HashMap<>();
+    headerMap.put("Content-Length", Integer.toString(lifeCycle.length()));
+    Map<String, String> queryParamMap = new HashMap<>();
+    queryParamMap.put("lifecycle", "");
+    HttpResponse response = executePut(bucketName, null, headerMap, queryParamMap, lifeCycle, 0);
+    response.body().close();
+
+  }
+
+  /**
+   * Delete the LifeCycle of bucket.
+   *
+   * @param bucketName   Bucket name.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code  deleteBucketLifeCycle("my-bucketname"); }</pre>
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException    upon requested algorithm was not found during
+   *                                     signature calculation
+   * @throws InsufficientDataException   upon getting EOFException while reading given
+   *                                     InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException         upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   */
+  public void deleteBucketLifeCycle(String bucketName)
+          throws InvalidBucketNameException, NoSuchAlgorithmException,
+          InsufficientDataException, IOException, InvalidKeyException, NoResponseException,
+          XmlPullParserException, ErrorResponseException, InternalException {
+    Map<String,String> queryParamMap = new HashMap<>();
+    queryParamMap.put("lifecycle", "");
+    HttpResponse response = executeDelete(bucketName,  "", queryParamMap);
+    response.body().close();
+  }
+
+
```

Comment:
We can remove the extra empty lines in middle

Reply:
removed.